server/container_create_linux.go

```diff
@@ -452,17 +453,39 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 	}
 
 	if ctr.Privileged() {
+		sysMountOptions := []string{"nosuid", "noexec", "nodev", "rw"}
+		sysFsCgroupMountOptions := []string{"nosuid", "noexec", "nodev", "rw", "relatime"}
+
+		runtimeHandler := sb.RuntimeHandler()
+		runtimeType, err := s.Runtime().RuntimeType(runtimeHandler)
+		if err != nil {
+			return nil, err
+		}
+
+		// A container is kernel separated if we're using shimv2, or we're using a kata v1 binary
+		podIsKernelSeparated := runtimeType == libconfig.RuntimeTypeVM ||
+			strings.Contains(strings.ToLower(runtimeHandler), "kata") ||
+			(runtimeHandler == "" && strings.Contains(strings.ToLower(s.config.DefaultRuntime), "kata"))
```

Comment:
can we add a method to the runtime class s.Runtime().IsKernelSeparated() instead of duplicating this check between here and sandbox_run_linux?

Reply:
Sure thing!